### PR TITLE
Improvements to oidc-middleware README

### PR DESCRIPTION
**Pull request by robertjd**
* Show use of profile scope by default
* Clarify the customization options
* Describe what this library is for
* A few other things I noticed

### CODE REVIEW

File: packages/oidc-middleware/README.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 
 ```sh
-npm install @okta/oidc-middleware
+npm install --save @okta/oidc-middleware
 ```
 
 ## ExpressOIDC API
 
-This module makes it easy to get users logged in to your Express app.
+It is assumed that you have already created an Okta Developer Org and created the necessary OIDC application in your Org.  If you are new to Okta or this flow, we suggest following the [Express.js Quickstart][express-quickstart].
```
Comment (lboyette-okta) [inline packages/oidc-middleware/README.md]: Should this info go under Prerequisites?

File: packages/oidc-middleware/README.md [hunk 2]
```diff
@@ -47,22 +47,24 @@ const oidc = new ExpressOIDC({
...
+* **client_secret** - A secret provided when you create an OIDC app in your Okta Org
 * **redirect_uri** - The callback for your app. Locally, this is usually `http://localhost:3000/authorization-code/callback`. When deployed, this should be `https://YOUR_PROD_DOMAIN/authorization-code/callback`.
 
 Optional config:
 
 * **response_type** - Defaults to `code`
-* **scope** - Defaults to `openid`
-* **routes** - Allows customization of the generated routes. See [#customizing-routes](#customizing-routes) for details.
+* **scope** - Defaults to `openid`, which will only return the `sub` claim. To obtain more information about the user, use `openid profile`. For a list of scopes and claims, please see [Scope-dependent claims](https://developer.okta.com/standards/OIDC/index.html#scope-dependent-claims-not-always-returned) for more information.
+
```
Comment (lboyette-okta) [inline packages/oidc-middleware/README.md]: nit: remove extra space

File: packages/oidc-middleware/README.md [hunk 3]
```diff
@@ -103,17 +105,17 @@ oidc.on('error', err => {
 });
 ```
 
-### oidc.ensureAuthenticated(redirectTo)
+### oidc.ensureAuthenticated({ redirectTo?: '/uri' })
```
Comment (lboyette-okta) [inline packages/oidc-middleware/README.md]: is this to indicate that it's optional?

File: packages/oidc-middleware/README.md [hunk 4]
```diff
@@ -154,34 +156,32 @@ app.get('/', (req, res) => {
...
     }
   }
 });
 ```
 
-* **path** - where our middleware attaches
-* **handler** - additional middleware for after we validate the OpenId Connect response
-* **defaultRedirect** - where we redirect to after login when there's no protected route to return to
+* **`callback.defaultRedirect`** - Where the user is redirected to after a successful authentication callback, if no `returnTo` value was specified by `oidc.ensureAuthenticated()`. Defaults to `/`.
+* **`callback.handler`** - A function that is called after a successful authentication callback, but before the final redirect within your application.  Useful for requirements such as conditional post-authentication redirects, or sending data to logging systems.
```
Comment (lboyette-okta) [inline packages/oidc-middleware/README.md]: nit: `application.  Useful` => `application. Useful`

File: packages/oidc-middleware/README.md [hunk 5]
```diff
@@ -154,34 +156,32 @@ app.get('/', (req, res) => {
...
+* **`callback.path`** - The URI that this library will host the callback handler on.  Defaults to `/authorization-code/callback`
```
Comment (lboyette-okta) [inline packages/oidc-middleware/README.md]: nit: `handler on.  Defaults` => `handler on. Defaults`

File: packages/oidc-middleware/README.md [hunk 6]
```diff
@@ -154,34 +156,32 @@ app.get('/', (req, res) => {
...
+* **`login.path`** - The URL that redirects the user to the authorize endpoint.  Defaults to `/login`.
```
Comment (lboyette-okta) [inline packages/oidc-middleware/README.md]: are we using URL or URI?

`endpoint.  Defaults` => `endpoint. Defaults`

File: packages/oidc-middleware/README.md (hunk 3 above)
Comment (robertjd) [reply packages/oidc-middleware/README.md]: Yeah, I'm open to other ideas?

File: packages/oidc-middleware/README.md [hunk 7]
```diff
@@ -1,44 +1,73 @@
...
   app.listen(3000, () => console.log('app started'));
 });
 oidc.on('error', err => {
   // An error occurred while setting up OIDC
-})
+});
 ```
 
-> By default, the session middleware uses a MemoryStore, which is not designed for production use. Use [another session store](https://github.com/expressjs/session#compatible-session-stores) for production.
+>
```
Comment (lboyette-okta) [inline packages/oidc-middleware/README.md]: nit: didn't affect the output, but this should be removed

**Pull request merged**